### PR TITLE
Synchronization issue solved without hotfix

### PR DESCRIPTION
**Pull request by nahuel-coliva**
Some regression_inputs.copy() may not be needed but are not harmful either, hotfix left only in debug mode (but actual operation left commented), added an .ini file (although removable, it's hyperopt+sfs with XGBoost instead of DT)

### CODE REVIEW

File: model_building/design_space.py [hunk 1]
```diff
@@ -286,25 +287,25 @@ def generate_experiment_configurations(self, prefix, regression_inputs):
...
-                                                       regression_inputs, prefix)
+                                                       regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.NNLS:
                 point = nnls.NNLSExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                         regression_inputs, prefix)
+                                                         regression_inputs.copy(), prefix)
             elif self._technique == ec.Technique.STEPWISE:
                 point = sw.StepwiseExperimentConfiguration(self._campaign_configuration, hyperparams_point_values,
-                                                           regression_inputs, prefix)
+                                                           regression_inputs.copy(), prefix)
```
Comment (brunoguindani) [inline model_building/design_space.py]: For readability's sake, I'd rather have a declaration of another variable (or an update of `regression_inputs`?) which is then passed in all these if-clauses, instead of a copy being produced on the spot in multiple lines

File: model_building/model_building.py [hunk 2]
```diff
@@ -141,9 +148,13 @@ def process(self, campaign_configuration, regression_inputs, processes_number):
...
         #        get out-of-synch from this moment on. In particular, it looks like wrapped x_columns of every expconf 
         #        are modified by each set_x_column() call, since they all take the value of the last set configuration
+        #UPDATE: No longer needed, but messages are left here for debug
+        self._logger.debug("After training, features look like:")
         for exp in expconfs:
+            self._logger.debug("\nWrapper: "+str(exp.get_x_columns()))
             if exp.is_wrapper():
-                exp._wrapped_experiment_configuration._regression_inputs = exp._regression_inputs
+                self._logger.debug("Wrapped: "+str(exp._wrapped_experiment_configuration.get_x_columns()))
+                #exp._wrapped_experiment_configuration._regression_inputs = exp._regression_inputs #old hotfix
```
Comment (brunoguindani) [inline model_building/model_building.py]: Good, I'd leave the debug messages and remove the unneeded lines

File: sequence_data_processing.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def __init__(self, input_configuration, debug=False, output="output", j=1, gener
             # Read configuration from the argument dict
             self.input_configuration_file = None
             self._campaign_configuration = input_configuration
-            general_args = {'output': output, 'seed': seed, 'j': j, 'debug': debug,
+            general_args = {'output': output, 'j': j, 'debug': debug,
```
Comment (brunoguindani) [inline sequence_data_processing.py:136]: Just confirming: this was removed because the `seed` variable is not defined here, correct?

File: example_configurations/bug_conf.ini [hunk 4]
```diff
@@ -0,0 +1,53 @@
+[General]
```
Comment (brunoguindani) [inline example_configurations/bug_conf.ini]: Why is this file needed? Should we replace `faas_test_hyperopt_sfs.ini` with this? What can we test with this file that we cannot with the other?


Long story short, I'd keep only one of them since they appear to be redundant. Let's choose which one

File: example_configurations/bug_conf.ini [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+max_features = ['auto']
+min_samples_split = ['loguniform(0.01,0.5)']
+min_samples_leaf = [1]
+
+[SVR]
+C = ['loguniform(0.001,1)']
+epsilon = ['loguniform(0.01,1)']
+gamma = [1e-7]
+kernel = ['linear']
+degree = [2]
```
Comment (brunoguindani) [inline example_configurations/bug_conf.ini]: If we do keep this, please fix the missing newline

**Review by brunoguindani — COMMENTED** [review]
Glad to see this finally works! See comments below.

File: model_building/design_space.py (hunk 1 above)
Comment (nahuel-coliva) [reply model_building/design_space.py]: You're right! After some more investigation I've found out that the problem could also be solved in the experiment configuration class, as you can see from the next commit

File: model_building/model_building.py (hunk 2 above)
Comment (nahuel-coliva) [reply model_building/model_building.py]: Modified

File: sequence_data_processing.py (hunk 3 above)
Comment (nahuel-coliva) [reply sequence_data_processing.py:136]: Definitely: it was causing the program to crash since it was not updated as the from-string configuration version

File: example_configurations/bug_conf.ini (hunk 4 above)
Comment (nahuel-coliva) [reply example_configurations/bug_conf.ini]: I used it to have a feedback when the xgboost was training, not sure why though: now removed

**Review by brunoguindani — APPROVED** [review]
Good :)
